jsscripts/embedhelper.js

```diff
@@ -330,15 +330,18 @@ EmbedHelper.prototype = {
             historyEntry.setURI(uri);
             shist.addEntry(historyEntry, true);
         });
-        webNav.sessionHistory.getEntryAtIndex(aMessage.data.index, true);
+        let index = aMessage.data.index;
+        if (index >= 0 && index < webNav.sessionHistory.count) {
+          webNav.sessionHistory.getEntryAtIndex(index, true);
```

Comment:
Without this call the back and forward buttons will be pointing at wrong history position. I guess this can be a bit alleviated with a couple of other conditions:

```
if (index < 0) {
    webNav.sessionHistory.getEntryAtIndex(0, true);
} else if (index >= webNav.sessionHistory.count) {
    webNav.sessionHistory.getEntryAtIndex(webNav.sessionHistory.count-1, true);
}
```
Also a warning message about index and history size mismatch would help to identify a possible error in browser UI code.